### ISSUE TITLE
Custom Theme with Polymer, not working

### ISSUE THREAD

**Issue by raffaeu** [mkdocs/mkdocs#1228]
Hi guys, I want to contribute to mkdocs so I started to create a new Theme using Polymer 2.0 and ES6.
The theme structure is the following:

```
- root folder
   - docs
   - my_theme
      - main.html
      - bower.json
      + bower_elements
   - mkdocs.yml
```

Then inside my main.html I started the `import` of CSS, JS and HTML files using the classic HTML5 syntax:

```html
  <head>
    <!-- Page title -->
    <title>{% if page.title %}{{ page.title }} - {% endif %}{{ config.site_name }}</title>
    <!-- Polymer 2.0 -->
    <script src="{{ base_url }}/bower_components/webcomponentsjs/webcomponents-loader.js"></script>
    <link rel="import" href="{{ base_url }}/bower_components/polymer/polymer.html">
    <!-- Main App -->
    <link rel="import" href="{{ base_url }}/src/main-app.html">
  </head>
```

But I get an error with `mkdocs serve` which is relative to my .html files. I get a **404 not found**. So I made a further investigation by running `mkdocs build --clean --verbose` and I noticed that inside the folder **bower_elements** only the `.js` and `.css` files are compiled inside the output. Any `.html` file is skipped. Why? They are *web components*, why don't you include them?

**Comment by raffaeu** [comment]
I found that, in booswatch, the creator has implemented a custom script in python that lets him copy to the output dir all files relative to a theme:
https://github.com/mkdocs/mkdocs-bootswatch/blob/master/setup.py

My question is, how far can I got with the script `setup.py`? Can I instruct the script to copy everything from my theme, including the `.html` and `.less` files, for example?

**Comment by waylan** [comment]
All `.html` files are assumed to be templates. Of course, templates should not get "copied" to the `site_dir`; they need to be run through the template engine first. The trick is telling MkDocs about those templates. There are a number of ways this happens:

* All Markdown files are passed to the `main.html` template by default.
* If a Markdown page defines a `template` meta-data value, then that template is used for that page instead of the default `main.html` (this is currently undocumented).
* Some templates are hard coded into MkDocs, like `search.html`.
* With the `extra_templates` config setting, users can include templates in their `docs_dir` and have MkDocs render them.

What is missing from that list is a way for a theme to include additional templates (essentially `extra_templates` for themes). In fact, in #1201, `static_templates` were introduced for that very purpose, which will be available when version 1.0 is released. See the updates to the documentation in that PR for details.

Finally, as `.html` files are templates, we can't just do a wholesale copy of all "unknown" `.html` files.  Consider template files which get "included" such as `toc.html`. They are never called directly by MkDocs, but are included by parent templates. Therefore, they are templates that MkDocs doesn't necessarily know about. My point is that there is no way for MkDocs to know the different between a plain HTML file and a template. Therefore, all `.html` files are assumed to be templates. We will not be making any changes in that regard.

That said, it may be possible to write a custom Plugin which addresses your need. Like the theme changes mentioned above, Plugin support will be available in version 1.0. For more info see #1223.

**Comment by raffaeu** [comment]
Well, thank you first for the exhaustive answer @waylan.
I do also understand the choice considering that `mkdocs` already has a template engine itself.
My idea of using Polymer 2.0 with ES6 was to give to `mkdocs` support to web components and shadow dom techniques.
I am curious about the plugin architecture especially because mkdocs became our primary documentation engine so I am willing to dive deeper into the architecture of the framework.

I'll have a look.
Thanks
